packages/engine/src/datastore/table/pool/mod.rs

```diff
@@ -74,12 +74,32 @@ impl<P: Pool<B> + Send + Sync, B: Batch> BatchPool<B> for P {
         self.get_batches_mut().push(Arc::new(RwLock::new(batch)))
     }
 
-    fn remove(&mut self, index: usize) -> Result<BatchReadProxy<B>> {
-        BatchReadProxy::new(&self.get_batches_mut().remove(index))
+    fn remove(&mut self, index: usize) -> Result<String> {
+        let mut batch_arc = self.get_batches_mut().remove(index);
+        if let Some(rw_lock) = Arc::get_mut(&mut batch_arc) {
+            // This can't deadlock as we just checked that the Arc owning this RwLock is unique
+            let batch = rw_lock.write();
+            Ok(batch.get_batch_id().to_string())
+        } else {
```

Comment:
If the `Arc::get_mut` failed, do we want to re-add the batch?

Reply:
No, it simply gives you the `&mut Vec`. `remove(index)` will give you the `Arc` which could have been cloned before.

Reply:
As we want to rework the logic anyway, I'd just `panic!`

Reply:
I think in all ways we'd use this function, this should be a fatal error, so yeah lets panic